### ISSUE TITLE
FTfunds possibly rejecting funds that have a colon in the symbol/code

### ISSUE THREAD

**Issue by cellarmation** [finance-quote/finance-quote#344]
I am using Finance::Quote via GnuCash to retrieve some quotes from the "Financial Times Funds service, GB", which I believe uses Ftfunds under the hood. This worked at some point in the past, but now fails for me.

Take this fund for example (Note that it can be accessed via just GB0006685613, but it gets redirected appending :GBX):
https://markets.ft.com/data/funds/tearsheet/summary?s=GB0006685613

- If I try to get a quote from this fund via GnuCash using a symbol of GB0006685613 I get the following error: "Error failed to find a currency"

- If I try to get a quote from this fund via GnuCash using a symbol of GB0006685613:GBX I get the following error: "Error - invalid symbol"

I have looked into this a bit and have a guess at what is going on (I have not looked at the FT API or Finance::Quote source before so may be confused). I suspect the fund now has to be retrieved via the longer symbol name that includes the colon now, or it isn't getting found, but when it is found Quote::Finance doesn't like that the symbol includes the colon when decoding the returned data.

I think the symbol returned in the data payload via the API now includes a colon as I used the example FT API [here](https://markets.ft.com/research/webservices/securities/v1/docs) and observed it for the above example fund.

Looking at the source [here](https://github.com/finance-quote/finance-quote/blob/e7d2a3939e9fab8c19156b1e6a44ff888045f63c/lib/Finance/Quote/FTfunds.pm) I suspect [this regex](https://github.com/finance-quote/finance-quote/blob/e7d2a3939e9fab8c19156b1e6a44ff888045f63c/lib/Finance/Quote/FTfunds.pm#L102C25-L102C25) isn't expecting a colon in the symbol field.

**Comment by bpschuck** [comment]
Cannot replicate issue with F::Q v1.58.

Using the `stockdump.pl` example script included with the module (Usually found where CPAN caches builds. On Linux/OSX it is usually `$HOME/.cpan/build/...`. On Windows it may be in `%HOMEPATH%\Appdata\Local\.cpan\...` or `C:\Strawberry\cpan\...`.):

```
bschuck@valhalla:~/.cpan/build/Finance-Quote-1.58-0$ ./Examples/stockdump.pl ftfunds GB0006685613
$VAR1 = {
          'GB0006685613method' => 'ftfunds',
          'GB0006685613date' => '09/08/2023',
          'GB0006685613isodate' => '2023-09-08',
          'GB0006685613source' => 'https://markets.ft.com/data/funds/tearsheet/summary?s=GB0006685613',
          'GB0006685613last' => '9.147',
          'GB0006685613net' => '0',
          'GB0006685613name' => 'Family Charities Ethical Trust Acc, GB0006685613:GBX',
          'GB0006685613success' => 1,
          'GB0006685613currency' => 'GBP',
          'GB0006685613time' => '17:00',
          'GB0006685613price' => '9.147',
          'GB0006685613errormsg' => 'Warning - failed to find a time',
          'GB0006685613nav' => '9.147',
          'GB0006685613symbol' => 'GB0006685613',
          'GB0006685613p_change' => '-0.00'
        };
```

Using `gnc-fq-dump` from GnuCash 4.8:

```
bschuck@valhalla:~/.cpan/build/Finance-Quote-1.58-0$ gnc-fq-dump ftfunds GB0006685613
Finance::Quote fields Gnucash uses:
    symbol: GB0006685613         <=== required
      date: 09/08/2023           <=== recommended
  currency: GBP                  <=== required
      last: 9.147                <=\
       nav: 9.147                <=== one of these
     price: 9.147                <=/
  timezone:                      <=== optional
```

As written, the original author uses the redirect to determine the currency and the `:GBX` or other such suffix should not be supplied in the symbol.

**Comment by cellarmation** [comment]
I am using GnuCash 5.3 with Finance::Quote 1.54. Not had much experience with Perl, but opened the interpreter and ran the same command. I got:
`
C:\Strawberry\cpan\build\Finance-Quote-1.54-0\Examples>stockdump.pl  ftfunds GB0006685613
$VAR1 = {
          'GB0006685613∟name' => 'Family Charities Ethical Trust Acc, GB0006685613:GBX',
          'GB0006685613∟errormsg' => 'Error - failed to find a currency',
          'GB0006685613∟success' => 0,
          'GB0006685613∟source' => 'https://markets.ft.com/data/funds/tearsheet/summary?s=GB0006685613',
          'GB0006685613∟symbol' => 'GB0006685613'
        };
`
I will try updating. I had forgotten that Finance::Quote is installed independently to GnuCash so had assumed it was up to date.

**Comment by cellarmation** [comment]
I have just updated, and can confirm the issue is resolved in 1.58. Thanks!

**Comment by bpschuck** [comment]
Thanks.
